scipy/stats/_morestats.py

```diff
@@ -3332,7 +3342,7 @@ def wilcoxon(x, y=None, zero_method="wilcox", correction=False,
             prob = np.sum(pmf[:r_plus + 1])
         prob = np.clip(prob, 0, 1)
 
-    return WilcoxonResult(T, prob, zstatistic=np.nan if mode == 'exact' else z)
+    return WilcoxonResult(T, prob, zstatistic=None if mode == 'exact' else z)
```

Comment:
Hmmm... I'm guess you did this so that for 1D input, which simply calls the function, `zstatistic` is `None`.
However, what happens for ND input in which `x.shape[axis] > 50` but there are lots of NaNs in some axis-slices (i.e. the length of some axis-slices < 50)? Does `zstatistic` become an object array with `None`s or do these turn into NaNs?

Reply:
The None is converted into a `nan`:

```py
In [1]: import numpy as np

In [2]: from scipy import stats

In [3]: x = np.random.rand(80, 10)

In [4]: x[:, 0] = np.nan

In [5]: stats.wilcoxon(x)
Out[5]: 
WilcoxonResult(statistic=array([nan,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.]), pvalue=array([           nan, 7.84952655e-15, 7.84952655e-15, 7.84952655e-15,
       7.84952655e-15, 7.84952655e-15, 7.84952655e-15, 7.84952655e-15,
       7.84952655e-15, 7.84952655e-15]), zstatistic=array([        nan, -7.76998525, -7.76998525, -7.76998525, -7.76998525,
       -7.76998525, -7.76998525, -7.76998525, -7.76998525, -7.76998525]))

In [6]:
```